duckduckgo/iOS
# Cookie prompt management

Task/Issue URL: https://app.asana.com/0/72649045549333/1201892492219034/f

**Description**:
- Support automatic cookie prompt management via autoconsent script
- Additionally hide cookie banners without reject button
- Present animated address bar notification whenever cookie pop-up is managed or hidden
- Promote the feature via Dax dialog on first encountered cookie pop-up by describing the feature and enabling opt-in
- Provide settings screen for the feature with toggle for its state and informative description
- Whenever cookie pop-up is managed or hidden show it in the privacy dashboard

**Additional notes:**
- list of websites for testing can be found in [Coverage testing](https://app.asana.com/0/0/1203843590207742/f)
- promoting Dax dialog is set to appear only for EU (checked via region of the phone set in iOS: Settings → General → Language & Region → Region)
- you can reset autoconsent settings via debug menu to restore default for the new app without the need of uninstalling it (also remember to clear cookies via fire button)
- promoting Dax dialog won't appear during onboarding

**Steps to test this PR**:
I. Dax dialog
1. Reinstall or reset autoconsent settings via debug menu
2. Ensure device's region is set to one in EU
3. Navigate to a page with cookie popup e.g. https://doodle.com/
4. Dax dialog for opt-in should appear
5. Test if selecting options adjusts settings

II. Cookies managed
1. When navigating for the first time to a page with cookie pop-up if feature is enabled a "Cookies managed" animated notification in the address bar should be presented following blocked trackers animation.
2. Privacy Dashboard should display an info about cookie pop-up

III. Pop-up Hidden
1. When navigating for the first time to a page with cookie pop-up that cannot be managed but hidden and if feature is enabled should present "Pop-up Hidden" animated notification in the address bar should be presented following blocked trackers animation.
2. Privacy Dashboard should display an info about cookie pop-up

IV. No Dax dialog outside EU
1. Reinstall or reset autoconsent settings via debug menu
2. Ensure device's region is NOT set to one in EU
3. Navigate to a page with cookie popup
4. Dax dialog should not be presented and the cookie pop-up shouldn't be dismissed (feature disabled by default)
5. Go to Settings -> Manage Cookie Pop-ups
6. Enable the feature with toggle
7. Cookie pop-ups should now be managed without Dax dialog being presented

V. No Dax dialog if feature was toggled manually in settings
1. Reinstall or reset autoconsent settings via debug menu
2. Ensure device's region is set to one in EU
3. Go to Settings -> Manage Cookie Pop-ups and enable the feature with toggle
4. Navigate to a page with cookie popup
5. Dax dialog should not be presented and the cookie pop-up should be dismissed

**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
